core/src/main/java/org/elasticsearch/index/query/IndexQueryParserService.java

```diff
@@ -125,6 +125,13 @@ public void setTermsLookupFetchService(@Nullable  TermsLookupFetchService termsL
         this.termsLookupFetchService = termsLookupFetchService;
     }
 
+    /**
+     * For testing purposes only - ensures that e.g. for GeoBoundingBoxQueries we can inject a mocked service.
+     * */
+    void setFieldDataService(IndexFieldDataService fieldDataService) {
+        this.fieldDataService = fieldDataService;
+    }
```

Comment:
the mock service could also be extended in the future if needed for other tests I guess